File: backend/packages/users/src/routes/get.py
from ..utils.route import BaseRoute
from ..utils.service import BaseService
from ..utils.request import Request
from ..utils.response import Response


class GetRoute(BaseRoute):
    _service: BaseService

    def __init__(self, service: BaseService) -> None:
        self._service = service

    def call(self, request: Request):
        user_id = request.data.get("userId", "")

        response = self._service.call(user_id=user_id)

        return Response(
            status_code=200,
            action=request.action,
            request_id=request.request_id,
            data=response,
        )

File: backend/packages/visualisations/src/routes/get.ts
import { BaseRoute } from '../utils/route'
import { BaseService } from '../utils/service'
import { makeResponse } from '../utils/response'

type Input = {
    connectionId: string
    projectId: string
    visualisationId: string
}

interface GetRoute extends BaseRoute<Input, any> { }

type Self = {
    service: BaseService<Input, any>
}

export const makeGetRoute = (service: BaseService<Input, any>): GetRoute => {
    const self: Self = {
        service
    }

    return {
        call: call(self)
    }
}

const call = (self: Self): GetRoute['call'] => async (request) => {
    const connectionId = request.connectionId
    const projectId = request.data.projectId
    const visualisationId = request.data.visualisationId

    const response = await self.service.call({ connectionId, projectId, visualisationId })

    return makeResponse(
        200,
        request.action,
        request.requestId,
        response
    )
}
